### DISCUSSION TITLE
Toggle and stepper/Progress bar

### DISCUSSION THREAD

**Discussion by ConsuelodeIruarrizaga** [twilio-labs/paste#2463] [Q&A] [answered]
Hi everyone! 
I am working on some screens for DialogFlow CX one-click, and we are using a stepper and a toggle. But these components aren't available on https://paste.twilio.design/. When do you think this component will be available? The frontend engineer is developing the screens now. 

PS 1: The toggle is available in the Figma library but not in Paste. Is this ok?
PS 2: I am using the stepper shared by Katie [Figma link ](https://www.figma.com/file/K1ftrUI2O7VYKZcKH2Ghdl/Stepper-component-brainstorming?node-id=142%3A22842)

DialogFlow CX one-click [Figma Link](https://www.figma.com/file/d54HaPcur2mamAwvdx0m1C/2022-Q2-DialogFlow-CX----One-Click-MVP?node-id=141%3A87616)

 Some images: 

<img width="454" alt="Screenshot 2022-06-13 at 10 51 03" src="https://user-images.githubusercontent.com/101562667/173316958-dbce76ac-ea44-41da-8328-dbf5f63a4ea5.png">

<img width="940" alt="Screenshot 2022-06-13 at 10 50 59" src="https://user-images.githubusercontent.com/101562667/173316968-7ec00430-24eb-4bb5-9cde-bb22421b7de8.png">

**Comment by kate-porter** [comment, marked as answer]
Hey @ConsuelodeIruarrizaga, thanks for asking about this! The Toggle component and the Stepper component are both high up on our roadmap, and design work is just about finalized, but we are unable to commit to a release date for the coded components at this time due to our [current focus on resiliency](https://github.com/twilio-labs/paste/discussions/2370). You can keep an eye on our [roadmap](https://paste.twilio.design/roadmap/) to stay up to date on what's coming in future releases (though, do note that any release marked "candidate" is not finalized). We'll be sure to announce the release of these components far and wide so you're in the loop once they're available!

In the meantime, your engineering team would need to build these custom using Paste primitives, and then plan to swap to the components once they're available.

Alternatively, if you want to avoid design/tech debt, you could also consider exploring alternatives to the Toggle and Stepper that already exist in Paste. For example, the Toggle could potentially be replaced by a Radio Group, like so:
![image](https://user-images.githubusercontent.com/5374045/173453526-428ed280-4189-488f-a342-e32ed25d86ec.png)

Stepper is a tougher one to replace, but for a shorter flow like this, you might be able to get away with not having a stepper and instead communicate the number of steps in the headings alone, like so:
![image](https://user-images.githubusercontent.com/5374045/173454083-8a7a27a7-2479-4ec3-9c6c-822ccdb90678.png)
or:
![image](https://user-images.githubusercontent.com/5374045/173455150-da799d85-1d20-406e-a472-635723838a8d.png)

Ultimately totally up to you, but wanted to toss some ideas out there for you since we don't have release dates for Toggle or Stepper yet. Let me know if you have any questions on this!
